src/inmanta/server/services/compilerservice.py

```diff
@@ -444,16 +451,37 @@ async def _run(self, compile: data.Compile) -> None:
             )
         await asyncio.sleep(wait)
 
+        merge_candidates: List[data.Compile] = [
+            c
+            for c in await data.Compile.get_next_compiles_for_environment(compile.environment)
+            if not c.id == compile.id and CompilerService._compile_merge_key(c) == CompilerService._compile_merge_key(compile)
+        ]
+        force_update: bool = any(c.force_update for c in chain([compile], merge_candidates))
+        if force_update:
+            awaitables = [
+                c.update_fields(force_update=True) for c in chain([compile], merge_candidates) if c.force_update is not True
```

Comment:
I don't think it is a good idea to rewrite the other requests.

Reply:
Do you think it is acceptable to rewrite `compile` (the request we're going to pass to the runner)? Otherwise more changes will be required: either
- treat a request--possibly other than `compile`--for which `force_update == True` as the master and pass that to the runner.
- add a parameter to the runner to override `force_update`

Reply:
ah, that is why you update them. Adding a parameter might be easiest?

Reply:
I think passing another compile would be easiest, but I also think that might be confusing, so I agree that adding a parameter is the better of those options.